src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -957,6 +958,23 @@ private ClusteringIndexNamesFilter reduceFilter(ClusteringIndexNamesFilter filte
         }
 
         NavigableSet<Clustering<?>> toRemove = null;
+
+        DeletionInfo deletionInfo = result.deletionInfo();
+
+        if (deletionInfo.hasRanges())
+        {
+            for (Clustering<?> clustering : clusterings)
+            {
+                RangeTombstone rt = deletionInfo.rangeCovering(clustering);
+                if (rt != null && rt.deletionTime().deletes(sstableTimestamp))
+                {
+                    if (toRemove == null)
+                        toRemove = new TreeSet<>(result.metadata().comparator);
+                    toRemove.add(clustering);
+                }
+            } 
+        }
```

Comment:
This is required because there are row deletions that were converted to range tombstones in CASSANDRA-15369, and therefore aren't picked up by the logic in `isRowComplete()`?